### ISSUE TITLE
Including a shape on schema-defined computed links 

### ISSUE THREAD

**Issue by avedetvedea** [edgedb/edgedb#4949]
I couldn't find an issue for this `UnsupportedFeatureError: including a shape on schema-defined computed links is not yet supported `, so creating one as a feature request.

**Comment by avedetvedea** [comment]
I guess for the most cases, this would be useful with `stored computeds` https://www.edgedb.com/roadmap#caching (thinking of something like statistical computeds in a type obtained from traversing other types and combining some stuff from them)

**Comment by msullivan** [comment]
Elvis was trying to solve this a while back with a grand unification of computeds and aliases (https://github.com/edgedb/edgedb/tree/computables-are-aliases) that ran into a bunch of snags and has now hopelessly bitrotted.
It's probably worth revisiting fresh.

**Comment by avedetvedea** [comment]
Yeah, I think much (if not all) of the behaviour is achievable using globals and aliases

**Comment by avedetvedea** [comment]
@msullivan Can we get this feature when free object typings land? I couldn't really understand why this is prohibited so I'm probably mistaken by assuming that problem was type/cardinality inference
